README.md

```diff
@@ -59,7 +59,7 @@ Or, in case of just a `from` with the default destination, you can also use a `{
 |[`force`](#force)|`{Boolean}`|`false`|Overwrites files already in `compilation.assets` (usually added by other plugins/loaders)|
 |[`ignore`](#ignore)|`{Array}`|`[]`|Globs to ignore for this pattern|
 |`flatten`|`{Boolean}`|`false`|Removes all directory references and only copies file names.⚠️ If files have the same name, the result is non-deterministic|
-|[`transform`](#transform)|`{Function}`|`(content, path) => content`|Function that modifies file contents before copying|
+|[`transform`](#transform)|`{Function}`|`(content, path) => content`|Function that modifies file contents before copying. This function may return a Promise too.|
```

Comment:
```
|[`transform`](#transform)|`{Function\|Promise}`|`(content, path) => content`|Function or Promise that modifies file contents before copying|
```

Reply:
Adding an example for `{Promise}` would be highly appreciated (~L201+), but not mandatory :)

```diff
 ### `transform`
 
+ #### `{Function}`

 **webpack.config.js**
 ```js
 [
   new CopyWebpackPlugin([
     {
       from: 'src/*.png',
       to: 'dest/',
       transform (content, path) {
         return optimize(content)
       }
     }
   ], options)
 ]

+ #### `{Promise}`

+  **webpack.config.js**
+  ```js
+  [
+    new CopyWebpackPlugin([
+     {
+       from: 'src/*.png',
+       to: 'dest/',
+       transform (content, path) {
+         return Promise.resolve(optimize(content))
+       }
+    }
+   ], options)
+ ]

```